gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/pooled_grpc_asyncio.py.j2

```diff
@@ -0,0 +1,445 @@
+
+        else:
+            if client_cert_source_for_mtls and not ssl_channel_credentials:
+                cert, key = client_cert_source_for_mtls()
+                self._ssl_channel_credentials = grpc.ssl_channel_credentials(
+                    certificate_chain=cert, private_key=key
+                )
+
+        # The base transport sets the host, credentials and scopes
+        super().__init__(
+            host=host,
+            credentials=credentials,
+            credentials_file=credentials_file,
+            scopes=scopes,
+            quota_project_id=quota_project_id,
+            client_info=client_info,
+            always_use_jwt_access=always_use_jwt_access,
+            api_audience=api_audience,
+        )
+        self._quota_project_id = quota_project_id
+        self.channel_pool: List[aio.Channel] = []
+        for i in range(pool_size):
+            new_channel = type(self).create_channel(
+                self._host,
+                # use the credentials which are saved
+                credentials=self._credentials,
+                # Set ``credentials_file`` to ``None`` here as
+                # the credentials that we saved earlier should be used.
+                credentials_file=None,
+                scopes=self._scopes,
+                ssl_credentials=self._ssl_channel_credentials,
+                quota_project_id=self._quota_project_id,
+                options=[
+                    ("grpc.max_send_message_length", -1),
+                    ("grpc.max_receive_message_length", -1),
+                ],
+            )
+            self.channel_pool.append(new_channel)
+
+        # Wrap messages. This must be done after self.channel_pool is populated
+        self._prep_wrapped_messages(client_info)
+
+    def next_channel(self) -> aio.Channel:
+        """Returns the next channel in the round robin pool."""
+        # Return the channel from cache.
+        channel = self.channel_pool[self._next_idx]
+        self._next_idx = (self._next_idx + 1) % len(self.channel_pool)
+        return channel
+
+    async def replace_channel(self, channel_idx, grace=None, new_channel=None) -> aio.Channel:
+        """
+        Replaces a channel in the pool with a fresh one.
+
+        The `new_channel` will start processing new requests immidiately,
+        but the old channel will continue serving existing clients for `grace` seconds
+
+        Args:
+          channel_idx(int): the channel index in the pool to replace
+          grace(Optional[float]): The time to wait until all active RPCs are 
+            finished. If a grace period is not specified (by passing None for
+            grace), all existing RPCs are cancelled immediately.
+          new_channel(grpc.aio.Channel): a new channel to insert into the pool
+            at `channel_idx`. If `None`, a new channel will be created.
+        """
+        if channel_idx >= len(self.channel_pool) or channel_idx < 0:
```

Comment:
Would it work well enough here if we just use `channel_idx % len(self.channel_pool)`?

Reply:
Yeah that could be an option. I'd be a bit worried about accidentally changing channels I didn't intent to though